cmd/qgb/query/cmd.go

```diff
@@ -0,0 +1,229 @@
+	dssync "github.com/ipfs/go-datastore/sync"
+	"github.com/libp2p/go-libp2p"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	tmlog "github.com/tendermint/tendermint/libs/log"
+)
+
+func Command() *cobra.Command {
+	queryCmd := &cobra.Command{
+		Use:          "query",
+		Aliases:      []string{"q"},
+		Short:        "Query relevant information from a running QGB",
+		SilenceUsage: true,
+	}
+
+	queryCmd.AddCommand(
+		Signers(),
+	)
+
+	queryCmd.SetHelpCommand(&cobra.Command{})
+
+	return queryCmd
+}
+
+func Signers() *cobra.Command {
+	command := &cobra.Command{
+		Use:   "signers <nonce>",
+		Args:  cobra.ExactArgs(1),
+		Short: "Queries the QGB for attestations signers",
+		Long: "Queries the QGB for attestations signers. The nonce is the attestation nonce that the command" +
+			" will query signatures for. It should be either a specific nonce starting from 2 and on." +
+			" Or, use 'latest' as argument to check the latest attestation nonce",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			config, err := parseFlags(cmd)
+			if err != nil {
+				return err
+			}
+
+			// creating the logger
+			logger := tmlog.NewTMLogger(os.Stdout)
+			logger.Debug("initializing queriers")
+
+			ctx, cancel := context.WithCancel(cmd.Context())
+			defer cancel()
+
+			stopFuncs := make([]func() error, 0)
+			defer func() {
+				for _, f := range stopFuncs {
+					err := f()
+					if err != nil {
+						logger.Error(err.Error())
+					}
+				}
+			}()
+
+			// create tm querier and app querier
+			tmQuerier, appQuerier, stops, err := common.NewTmAndAppQuerier(logger, config.tendermintRPC, config.celesGRPC)
+			stopFuncs = append(stopFuncs, stops...)
+			if err != nil {
+				return err
+			}
+
+			// creating the host
+			h, err := libp2p.New()
+			if err != nil {
+				return err
+			}
+			addrInfo, err := peer.AddrInfoFromString(config.targetNode)
+			if err != nil {
+				return err
+			}
+			for i := 0; i < 5; i++ {
+				logger.Debug("connecting to target node...")
+				err := h.Connect(ctx, *addrInfo)
+				if err != nil {
+					logger.Error("couldn't connect to target node", "err", err.Error())
+				}
+				if err == nil {
+					logger.Debug("connected to target node")
+					break
+				}
+				time.Sleep(5 * time.Second)
+			}
+
+			// creating the data store
+			dataStore := dssync.MutexWrap(ds.NewMapDatastore())
+
+			// creating the dht
+			dht, err := p2p.NewQgbDHT(cmd.Context(), h, dataStore, []peer.AddrInfo{}, logger)
+			if err != nil {
+				return err
+			}
+
+			// creating the p2p querier
+			p2pQuerier := p2p.NewQuerier(dht, logger)
+
+			nonce, err := parseNonce(ctx, appQuerier, args[0])
+			if err != nil {
+				return err
+			}
+			if nonce == 1 {
+				return fmt.Errorf("nonce 1 doesn't need to be signed. signatures start from nonce 2")
+			}
+
+			err = getSignaturesAndPrintThem(ctx, logger, appQuerier, tmQuerier, p2pQuerier, nonce)
+			if err != nil {
+				return err
+			}
```

Comment:
how long should this take and how heavy is this operation? should we just query an existing relayer instead of spinning up one in memory and querying the dht?

Reply:
https://github.com/celestiaorg/orchestrator-relayer/issues/321

Reply:
okie dokie that's what I thought